examples/crossaccounts/README.md

```diff
@@ -0,0 +1,7 @@
+#  Example Transit Gateway that connects two VPCs in two different accounts
+
+This example creates 2 VPCs in two different accounts (or as alternative allows to use an existing VPC for the second account), and illustrates how to provision a transit gateway in the first of them and to configure connectivity between the 2 VPCs, performing the approval action.
+
+Two providers are defined for the two IBM Cloud accounts involved into the example.
+
+Due to a bug with provider at plan step https://github.com/IBM-Cloud/terraform-provider-ibm/issues/4445 the approval action must be performed when the resources are already created. This is why the example allows to run the same template twice with the input var `run_approval` set initially to *false* to skip the ibm_tg_connection_action resource and then again setting to *true* to run the approval step.
```

Comment:
It seems to be fixed on provider side. I asked them to release a new beta version at https://github.com/IBM-Cloud/terraform-provider-ibm/pull/4446

Reply:
yes, I am waiting for the next release as the fix is already merged
I will modify the test as soon as the fix is confirmed to work

Reply:
yes, I am waiting for the next release as the fix is already merged
I will modify the test as soon as the fix is confirmed to work

> Thanks @vbontempi - great PR, I put some comments inline. Would it be possible to also:
> 
>     1. Add a readme in the submodule (with generated terradoc)
> 
>     2. Update the readme at top level to indicate that there is also a submodule to approve connections
> 
>     3. Add an option in the parent module to optionally create a transit gateway (and take an existing transit gateway id as input like in the submodule). This adds support to add connections to an existing transit gateway (created outside terraform) and then approve them with submodule.

yes, I am ok with these changes to have the module usage more 'customisable' to different consumers' needs

> I'm not sure this is a breaking change here given the logic in a separate submodule - what makes you think it is?

The reason of the breaking change is the change I had to make to the output of the main module from a list of connectionIDs to a map of them, in order to be able to identify (through the vpc crn) the ones needing to be approved through the action submodule

This would break any consumer of the module using the output in their template

Reply:
@vburckhardt fix on provider side solved the bug, so the workaround is removed. Pushed final version